### PR TITLE
Show default mouse cursor over documentation panel

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

[Task link](https://www.pivotaltracker.com/story/show/183970777)

The default mouse cursor is now displayed over the documentation panel.

https://user-images.githubusercontent.com/6566674/205954491-177fcc01-7e15-4a5c-825e-c0b02be527cc.mp4

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Review by wdanilo — CHANGES_REQUESTED** [review]
The linked task does not mention native cursor. Only mentions that the cursor should be visible. I remember that very long time ago there was a reason why we did not display our normal cursor, but what is preventing us now from doing so?

**Comment by farmaazon** [comment]
> The linked task does not mention native cursor. Only mentions that the cursor should be visible. I remember that very long time ago there was a reason why we did not display our normal cursor, but what is preventing us now from doing so?

That we still need to display HTML documentation _over_ the node, so we:
1. display documentation panel over the canvas, thus covering our own cursor
2. display documentation panel under the canvas, cutting out all nodes over it, so only our pointer remains - but we have no mechanics of cutting portions of layers, making this task costly

The linked task mention the system cursor in its title. We have this also in the notes in https://www.pivotaltracker.com/story/show/180872953 (as an outcome of our discussion long ago).

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/180872953">New look of the components' documentation. - Pivotal Tracker</a></strong></div></blockquote>

**Comment by wdanilo** [comment]
Ohh, I forgot that the case is that EnsoGL doesnt support "cutting holes" yet and this is needed here. Ok, thank you so much for the reminder about it!

File: lib/rust/ensogl/core/src/application.rs [hunk 1]
```diff
@@ -41,6 +41,10 @@ pub mod traits {
 crate::define_endpoints_2! {
     Input {
         set_tooltip(tooltip::Style),
+        /// Show the default mouse cursor.
+        show_default_cursor(),
+        /// Hide the default mouse cursor.
+        hide_default_cursor(),
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/application.rs]: I would not call it the "default" cursor. Rather a "system cursor"

File: lib/rust/ensogl/core/src/application.rs [hunk 2]
```diff
@@ -75,6 +79,14 @@ pub struct ApplicationData {
     update_themes_handle: callback::Handle,
 }
 
+impl ApplicationData {
+    /// Show or hide the mouse cursor by setting the `cursor` CSS property of the `body` element.
+    fn show_cursor(&self, show: bool) {
+        let style = if show { "auto" } else { "none" };
+        web::document.body_or_panic().set_style_or_warn("cursor", style);
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/application.rs]: This should also have "system cursor" in name. As after hiding, our cursor will still be visible, and thus, this method might be a little bit misleading.

File: lib/rust/ensogl/core/src/application.rs [hunk 3]
```diff
@@ -109,7 +116,22 @@ impl Application {
...
+    fn init(self) -> Self {
+        let frp = &self.frp;
+        let data = &self.inner;
+        let network = self.frp.network();
+        enso_frp::extend! { network
+            frp.private.output.tooltip <+ frp.private.input.set_tooltip;
+            eval_ frp.private.input.show_default_cursor(data.show_cursor(true));
+            eval_ frp.private.input.hide_default_cursor(data.show_cursor(false));
+        }
+        // We hide the default cursor to replace it with the ensogl cursor.
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/application.rs]: minor: EnsoGL is a name. Also default- > system

**Review by wdanilo — APPROVED** [review]

**Pull request merged**